docs/docs/guides/guides/js_apps/rotate_keys.md

```diff
@@ -0,0 +1,40 @@
+---
+title: How to Rotate Nullifier Keys
+---
+
+This guide explains how to rotate nullifer secret and public keys using Aztec.js. To learn more about key rotation, read the [concepts section](../../../aztec/aztec/concepts/accounts/keys.md#key-rotation).
+
+## Prerequisites
+
+You should have a wallet whose keys you want to rotate. You can learn how to create wallets from [this guide](./create_account.md).
+
+You should also have a PXE initialized.
+
+## Relevant imports
+
+You will need to import these from Aztec.js:
+
+#include_code imports yarn-project/end-to-end/src/e2e_key_rotation.test.ts typescript
+
+## Create nullifier secret and public key
+
+`newNskM` = new nullifier secret key, master
+`newNpkM` = new nullifier public key, master (type `PublicKey`)
+
+#include_code create_keys yarn-project/end-to-end/src/e2e_key_rotation.test.ts typescript
+
+## Rotate nullifier secret and public key
+
+Call `rotateMasterNullifierKey` on the PXE to rotate the secret key. 
+
+#include_code rotateMasterNullifierKey yarn-project/end-to-end/src/e2e_key_rotation.test.ts typescript
+
+## Rotate public key 
+
+Create a key registry with your wallet. 
+
+#include_code keyRegistryWithB yarn-project/end-to-end/src/e2e_key_rotation.test.ts typescript
+
+Then `rotate_npk_m` on the key registry contract to rotate the public key:
+
+#include_code rotate_npk_m yarn-project/end-to-end/src/e2e_key_rotation.test.ts typescript
```

Comment:
nice - more as an fyi to you
All of our current contracts break if you do key rotation - https://github.com/AztecProtocol/aztec-packages/issues/6312